### PR TITLE
ADR 6: REST API

### PR DESCRIPTION
**Pull request by rabernat**
I am proposing we create a simple REST API to provide a central source of truth for Pangeo Forge Federation. Details in the PR.

comments welcome from @pangeo-forge/dev-team

### CODE REVIEW

**Comment by rabernat** [comment]
Just to make this more concrete. I would also propose the following stack to implement the API endpoint.

- [SQLModel](https://sqlmodel.tiangolo.com/) for the core data model (the classes and their relationships)
- [FastAPI](https://fastapi.tiangolo.com/) for the API routes
- [Heroku](https://www.heroku.com/) + [Heroku Postgres](https://www.heroku.com/postgres) for deployment. (We already have a tiny API for Pangeo Gallery deployed this way: https://github.com/pangeo-gallery/pangeo-gallery-bot)

File: doc/adr/0006_pangeo_forge_REST_API.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+- `GET /recipe_runs`
+
+And similar methods. We could also implement queries, etc.
+
+#### Datasets API
+
+We could also expose the datasets via the API.
+This of course overlaps with STAC stuff / catalogging stuff.
+It would be really cool to be able to track `dataset` ➡️ `recipe_run` ➡️ `feedstock`
```
Comment (martindurant) [inline doc/adr/0006_pangeo_forge_REST_API.md:95]: More that really cool - I'd say imperative!

**Comment by rabernat** [comment]
I also just read about [fastapi-stac](https://github.com/stac-utils/stac-fastapi). So in theory we could eventually include a STAC component in a more general fastapi service.

**Comment by rabernat** [comment]
Adding a note here about the DataOne API, which is an interesting point of reference:

https://releases.dataone.org/online/api-documentation-v2.0.1/apis/index.html

**Comment by abarciauskas-bgse** [comment]
closing as stale

**Pull request closed without merging**